lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -986,36 +966,17 @@ assignScriptRedeemers pparams timeTranslation utxo redeemers tx =
     --
     -- Redeemers are determined from the context given to the caller via the
     -- 'Redeemer' type which is mapped to an 'Alonzo.ScriptPurpose'.
-    assignNullRedeemersBabbage
-        :: BabbageTx
-        -> Either ErrAssignRedeemers (Map Alonzo.RdmrPtr Redeemer, BabbageTx)
-    assignNullRedeemersBabbage babbageTx = do
-        (indexedRedeemers, nullRedeemers) <- fmap unzip $ forM redeemers $ \rd -> do
-            ptr <-
-                case Alonzo.rdptr (Alonzo.body babbageTx) (toScriptPurpose rd) of
-                    SNothing -> Left $ ErrAssignRedeemersTargetNotFound rd
-                    SJust ptr -> pure ptr
-            rData <- case deserialiseOrFail (BL.fromStrict $ redeemerData rd) of
-                Left e -> Left $ ErrAssignRedeemersInvalidData rd (show e)
-                Right d -> pure (Alonzo.Data d)
-            pure ((ptr, rd), (ptr, (rData, mempty)))
-        pure
-            ( Map.fromList indexedRedeemers
-            , babbageTx
-                { Alonzo.wits = (Alonzo.wits babbageTx)
-                    { Alonzo.txrdmrs =
-                        Alonzo.Redeemers (Map.fromList nullRedeemers)
-                    }
-                }
+    assignNullRedeemers
+        :: Write.RecentEraLedgerConstraints (Write.ShelleyLedgerEra era)
+        => Write.Tx (Write.ShelleyLedgerEra era)
+        -> Either ErrAssignRedeemers
+            ( Map Alonzo.RdmrPtr Redeemer
+            , (Write.Tx (Write.ShelleyLedgerEra era))
             )
-
-    assignNullRedeemersConway
-        :: ConwayTx
-        -> Either ErrAssignRedeemers (Map Alonzo.RdmrPtr Redeemer, ConwayTx)
-    assignNullRedeemersConway conwayTx = do
+    assignNullRedeemers ledgerTx = do
         (indexedRedeemers, nullRedeemers) <- fmap unzip $ forM redeemers $ \rd -> do
             ptr <-
-                case Alonzo.rdptr (Alonzo.body conwayTx) (toScriptPurpose rd) of
+                case Alonzo.rdptr (Write.txBody (recentEra @era) ledgerTx) (toScriptPurpose rd) of
```

Comment:
Hi @Anviking!

Unfortunately this line is too long.
Luckily, we can fix it by extracting out the body of `forM` into a separate function:

See: 856af043a64f6fc24fc621909aa435573fd82179

Reply:
👍 thanks!